test/get/git/path_test.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+import 'package:pub/src/io.dart';
+
+import '../../descriptor.dart' as d;
+import '../../test_pub.dart';
+
+main() {
```

Comment:
I am also curious what happens if you have a package nested inside another package. I think its totally fine to not support that, but if it isn't supported lets make sure it throws a nice error.

For example, can you depend on both of these packages:
```
my_repo/
  pkg1/
    pubspec.yaml
    pkg2/
      pubspec.yaml
```